lib/wibox/hierarchy.lua

```diff
@@ -51,6 +51,15 @@ local function hierarchy_new(redraw_callback, layout_callback, callback_arg)
         end
         layout_callback(result, callback_arg)
     end
+    function result._emit_recursive(_, name, ...)
+        local cur = result
```

Comment:
I'd propose to do
```
function result._emit_recursive(self, name, ...)
  local cur = self
```
That way, this function still has to be called with the correct argument.